docs/existing-aws-infrastructure.md

```diff
@@ -0,0 +1,94 @@
+# Consuming Existing AWS Infrastructure
+
+Normally, Cluster API will create infrastructure on AWS when standing up a new workload cluster. However, it is possible to have Cluster API re-use existing AWS infrastructure instead of creating its own infrastructure. Follow the instructions below to configure Cluster API to consume existing AWS infrastructure.
+
+## Prerequisites
+
+In order to have Cluster API consume existing AWS infrastructure, you will need to have already created the following resources:
+
+* A VPC
+* One or more private subnets (subnets that do not automatically assign a public IP address to EC2 instances)
+* A public subnet in the same Availability Zone (AZ) for each private subnet (this is required for NAT gateways to function properly)
+* A NAT gateway for each private subnet, along with associated Elastic IP addresses
+* An Internet gateway for all public subnets
+* Route table associations that provide connectivity to the Internet through a NAT gateway (for private subnets) or the Internet gateway (for public subnets)
+
+You will need the following information, which can be obtained either via the AWS Management Console or using the AWS CLI:
+
+1. The ID of the VPC
+2. The IDs of the subnets in the VPC that you want Cluster API to use
```

Comment:
The subnet IDs are actually ignored; given a VPC we'll always look up all the subnets and populate them on the cluster object via [describe](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0b83f06314410a9a3d628f006a647086f51736a8/pkg/cloud/services/ec2/subnets.go#L170).

Providing the subnet IDs wouldn't bother CAPA, but it might lead the humans to believe they can ask CAPA to use a subset of IDs by providing a reduced list. At least at present, CAPA would ignore that intent.

Reply:
Fixed via 38ce860.

Reply:
TIL, and this seems...not ideal?

Reply:
I’ve tested it, and the subnet IDs are not required. I believe that text has been removed in the most recent commit against this PR.